### PR TITLE
Centralized build tools in `re_build_tools`

### PR DESCRIPTION
**Pull request by teh-cmc**
Centralize our build tools in one place (`re_build_tools`) and re-use those everywhere.

Cherry picked from #2329 which both re-uses existing tools adds new ones.

~Technically, `re_build_build_info` is not the correct place for this, and one could imagine yet another crate `re_build_tools` for these things; but for now the overhead of yet another crate doesn't seem worth it.~

### CODE REVIEW

File: examples/rust/objectron/build.rs [hunk 1]
```diff
@@ -54,6 +54,7 @@ fn main() -> Result<(), std::io::Error> {
 }
 
 /// Only touch the file if the contents has actually changed
+// TODO(cmc): use the same source tracking system as re_types*
 fn write_file_if_necessary(
```
Comment (emilk) [inline examples/rust/objectron/build.rs:58]: Move it into `re_build_build_info` as well?

**Review by emilk — APPROVED** [review]
looks good, but perhaps we should just rename `re_build_build_info` to `re_build_tool`

**Review by Wumpf — APPROVED** [review]
very nice. I wouldn't mind renaming `re_build_build_info` itself to `re_build_tools` without splitting

File: examples/rust/objectron/build.rs (hunk 1 above)
Comment (teh-cmc) [reply examples/rust/objectron/build.rs:58]: I was hesitant to have the objectron example depends on re_build_info... but then again I guess it's just the build-dependencies, let's go for it

File: Cargo.toml [hunk 2]
```diff
@@ -25,7 +25,7 @@ version = "0.7.0-alpha.0"
 # re_log_types 0.3.0-alpha.0, NOT 0.3.0-alpha.4 even though it is newer and semver-compatible.
 re_analytics = { path = "crates/re_analytics", version = "0.7.0-alpha.0", default-features = false }
 re_arrow_store = { path = "crates/re_arrow_store", version = "0.7.0-alpha.0", default-features = false }
-re_build_build_info = { path = "crates/re_build_build_info", version = "0.7.0-alpha.0", default-features = false }
+re_build_tools = { path = "crates/re_build_tools", version = "0.7.0-alpha.0", default-features = false }
```
Comment (emilk) [inline Cargo.toml:28]: nit: no longer sorted :)

**Pull request merged**